lib/takeaway_service.rb

```diff
@@ -0,0 +1,92 @@
+require_relative 'menu'
+require_relative 'dish'
+require_relative 'send_sms'
+
+class TakeawayService
+  attr_reader :basket
+
+  def initialize(menu_class = Menu.new, text_service_class = SendSms)
+    @menu_class = menu_class
+    @text_service_class = text_service_class
+    @basket = {}
+  end
+
+  def add_to_basket(dish, quantity = 1)
+    not_on_menu_message = "Cannot add to basket: dish not on the menu"
+    raise not_on_menu_message if dish_not_on_the_menu(dish)
+
+    incorrect_quantity_message = "Cannot add to basket: quantity not recognised"
+    raise incorrect_quantity_message if quantity_less_than_1(quantity)
+
+    if already_added?(dish)
+      add_quantity(dish, quantity)
+    else
+      add_selection(dish, quantity)
+    end
+    "#{quantity} #{dish}(s) added to your basket"
+  end
+
+  def view_menu
+    @menu_class.view
+  end
+
+  def order_summary
+    print order
+  end
+
+  def total_cost
+    print cost
+  end
+
+  def place_order
+    @text_service_class.new.message
+  end
+
```

Comment:
Some of your private methods look like they do exactly the same as the public ones (other than add the word print). Is it worth adding a new method just for that? It might help reduce your private methods by just doing the work in the public methods. It depends on what you want to go for.